src/Caster.Api/Features/Directories/Requests/GetAll.cs

```diff
@@ -36,18 +38,39 @@ public class Query : IRequest<Directory[]>
             public bool IncludeFileContent { get; set; }
         }
 
-        public class Handler(ICasterAuthorizationService authorizationService, IMapper mapper, CasterContext dbContext) : BaseHandler<Query, Directory[]>
+        public class Handler(
+            ICasterAuthorizationService authorizationService,
+            IMapper mapper,
+            CasterContext dbContext,
+            IIdentityResolver identityResolver) : BaseHandler<Query, Directory[]>
         {
-            public override async Task<bool> Authorize(Query request, CancellationToken cancellationToken) =>
-                await authorizationService.Authorize([SystemPermission.ViewProjects], cancellationToken);
+            public override Task<bool> Authorize(Query request, CancellationToken cancellationToken) => Task.FromResult(true);
 
             public override async Task<Directory[]> HandleRequest(Query request, CancellationToken cancellationToken)
             {
-                return await dbContext.Directories
-                    .Expand(mapper.ConfigurationProvider, request.IncludeRelated, request.IncludeFileContent)
-                    .ToArrayAsync();
+                if (await authorizationService.Authorize([SystemPermission.ViewProjects], cancellationToken))
+                {
+                    return await dbContext.Directories
+                        .Expand(mapper.ConfigurationProvider, request.IncludeRelated, request.IncludeFileContent)
+                        .ToArrayAsync();
+                }
+                else
+                {
+                    var userId = identityResolver.GetId();
+                    var myProjectIds = await dbContext.ProjectMemberships
+                        .Where(pm => pm.UserId == userId)
+                        .Select(pm => pm.ProjectId)
+                        .ToListAsync(cancellationToken);
+
+                    var myDirectories = await dbContext.Directories
+                        .Where(d => myProjectIds
+                        .Contains(d.ProjectId))
+                        .ProjectTo<Directory>(mapper.ConfigurationProvider)
```

Comment:
should this have the same .Expand(mapper.ConfigurationProvider, request.IncludeRelated, request.IncludeFileContent) as in the other case?

Reply:
yes.  change coming